### ISSUE TITLE
USA Eastern Interconnection

### ISSUE THREAD

**Issue by HansHyde** [tmrowco/electricitymap#903]
EIA on Overall North American Grid... https://www.eia.gov/todayinenergy/detail.php?id=27152

- 26+ Balancing Agencies (BA)
1. **ENTERGY** - (not their merchant nuclear fleet) - Louisiana & up the Mississippi River
2. **NE** - "Nebraska"???
3. **SPP N** - "Southwest Power Pool North"
4. **SPP S** - "Southwest Power Pool South"
5. **MAPP CA** - "MidAmerica Power Pool??? - Canada"
6. **MAPP US** - "MidAmerica Power Pool??? - US"
7. **MISO IN** - "Midwest ISO - Indiana"
8. **MISO MI** - "Midwest ISO - Michigan"
9. **MISO MO IL** - "Midwest ISO - Missouri & Illinois"
10. **MISO W** - "Midwest ISO - Wisconsin???"
11. **MISO WUMS** - "Midwest ISO - Wisconsin & Upper Michigan State (UpperPenninsula)"
12. **Non RTO Midwest** -
13. **HQ** - "Hydro Quebec"
14. **Maritimes** - "Maritime provinces of Canada"
15. **NEISO** - "New England ISO" - Parser #924 
16. **NYISO 1** (Zones A-F) - most of NY
17. **NYISO 2** (Zones GHI) - the downstate “wedge”
18. **NYISO 3** (Zones J&K) - NYC & Long Island
19. **IESO** - "Independent Electric System Operator" - Ontario - Parser #925 
20. **PJM Eastern MAAC** - "Pennsylvania, (New) Jersey, Maryland Eastern MAAC"
21. **PJM Rest of MAAC** - 
22. **PJM Rest of RTO** -
23. **FRCC** - "Florida RCC" - mainly FPL & Duke
24. **SOCO** - most likely "Southern Company" - owns Georgia Power, Alabama Power, Mississippi Power & Gulf Power
25. **TVA** - "Tennessee Valley Authority"
26. **VACA** - think this is the Carolinas
27. TBD - _eai.gov claims 36_
28. TBD

- 36 US States and 5 or 6 Canadian Provinces 
- **58 Known Interconnections** 
1. ENTERGY 	<—>     SPP N
2. ENTERGY 	<—>     	SPP S
3. ENTERGY	 	<—>     MISO MO IL
4. ENTERGY 	<—>     	SOCO
5. ENTERGY 	<—>     	TVA
6. ENTERGY	 	<—>     ERCOT
7. NE	 	<—>     SPP N
8. NE	 	<—>     MAPP US
9. NE	 	<—>     MISO W
10. NE	 	<—>     RMPA
11. SPP N	 	<—>     SPP S
12. SPP N	 	<—>     MISO MO IL
13. SPP N	 	<—>     MISO W
14. SPP N	 	<—>     RMPA
15. SPP S	 	<—>     ERCOT
16. SPP S	 	<—>     AZ NM SNV
17. MAPP CA	 	<—>     MAPP US
18. MAPP CA	 	<—>     MISO W
19. MAPP CA	 	<—>     IESO
20. MAPP US	 	<—>     MISO W
21. MAPP US	 	<—>     NWPP
22. MAPP US	 	<—>     RMPA
23. MISO IN	 	<—>     MISO MI
24. MISO IN	 	<—>     MISO MO IL
25. MISO IN	 	<—>     Non RTO Midwest
26. MISO IN	 	<—>     PJM Rest of RTO
27. MISO MI	 	<—>     MISO WUMS
28. MISO MI	 	<—>     IESO
29. MISO MI	 	<—>     PJM Rest of RTO
30. MISO MO IL	 	<—>     MISO W
31. MISO MO IL	 	<—>     PJM Rest of RTO
32. MISO MO IL	 	<—>     TVA
33. MISO W	 	<—>     MISO WUMS
34. MISO W	 	<—>     IESO
35. MISO W	 	<—>     PJM Rest of RTO
36. MISO WUMS	 	<—>     PJM Rest of RTO
37. Non RTO Midwest	 	<—>     TVA
38. HQ	 	<—>     NEISO
39. HQ	 	<—>     NYISO 1 (A-F)
40. HQ	 	<—>     IESO
41. Maritimes	 	<—>     NEISO
42. NEISO	 	<—>     NYISO 1 (A-F)
43. NEISO	 	<—>     NYISO 2 (GHI)
44. NEISO	 	<—>     NYISO 3 (J&K)
45. NYISO 1 (A-F)	 	<—>     NYISO 2 (GHI)
46. NYISO 1 (A-F)	 	<—>     IESO               

- [ ] RTD in BA#16 - NYISO

- [ ] RTD in BA#19 - Ontario

47. NYISO 1 (A-F)	 	<—>     PJM Rest of MAAC
48. NYISO 2 (GHI)	 	<—>     NYISO 3 (J&K)
49. NYISO 2 (GHI)	 	<—>     PJM Eastern MAAC
50. NYISO 3 (J&K)	 	<—>     PJM Eastern MAAC
51. PJM Eastern MAAC	 	<—>     PJM Rest of MAAC
52. PJM Rest of MAAC	 	<—>     PJM Rest of RTO
53. PJM Rest of RTO	 	<—>     TVA
54. PJM Rest of RTO	 	<—>     VACA
55. FRCC	 	<—>     SOCO
56. SOCO	 	<—>     TVA
57. SOCO	 	<—>     VACA
58. TVA	 	<—>     VACA

- Greater than 60% of USA/Canadian Power Generation & Load covered
- Represents on the order of 800,000 MW of installed generation capacity
- Only matches state line in 2 or 3 occasions (TVA, NYISO - has 3 BAs within, Florida - has 4 utilities within) - Canadian Provinces follow province lines
- In the US's enormously complicated "grid" there are thousands of regulated, unregulated, IOUs, MOUs, CoOps, REAs, DSOs, TSOs, ISOs, RTOs, BAs, merchant fleet generators, etc.  From the electric grid perspective, state lines only "connect" in a regulatory nature (state level Public Utilities Commissions) that will have little to no information regarding day-to-day or realtime grid operations.
- BAs (which NYISO, PJM, CAISO can be) appear to the most likely source (if available) to open access data for EM.org's objectives.

Here is a Flowchart for the Eastern Interconnection's BAs & Interfaces.... http://images.slideplayer.com/39/10987225/slides/slide_4.jpg

I have started an Excel file with the 26 BAs (parsers) & 58 Interfaces - happy to open up the Dropbox folder to those wanting access.. 

PS... is there a better way to use Github with the "Projects" for this???

**Comment by jarek** [comment]
Is there any real-time generation or fuel mix data?

**Comment by HansHyde** [comment]
@jarek We've already found NYISO (albeit IMHO the CO2 data is likely off by 20-50%).

I just found SPP... although I am not sure if their realtime info link showing on their page is something yet to be built/released or behind a non-US IP address firewall (which is stopping me from accessing ERCOT realtime info).  Check it out here... https://www.spp.org

**Comment by HansHyde** [comment]
@jarek We've already found NYISO (albeit IMHO the CO2 data is likely off by 20-50%).

I just found SPP... although I am not sure if their realtime info link showing on their page is something yet to be built/released or behind a non-US IP address firewall (which is stopping me from accessing ERCOT realtime info).

ps... not sure of your professional background, but what is written above represents something much larger than Europe... and unlike ENTSO??? that aggregates and supplies all the information EM is looking for in the EU, there is nothing like that in the US.  The only reason IESO & CAISO info has been relatively easy to locate - they both participate in a carbon cap & trade system.  No place else has that in the US - and if you haven't noticed Trump & his clowns (Pruitt & Perry) are setting up to give Billions of USD to the coal (and nuclear) entities.  We will be very lucky to find "generation mixes" for 60% of the above.  CO2 data - ha, fat chance!  But we might just be able to make a lot of very well informed guesstimates - and that's what I'm hoping!  If you guys can code, I can point you in the right directions. :)

**Comment by HansHyde** [comment]
@jacobbaratta @synfo fyi in regards to #143

**Comment by HansHyde** [comment]
[Eastern Interconnect Flowchart.xlsx](https://github.com/tmrowco/electricitymap/files/1555990/Eastern.Interconnect.Flowchart.xlsx)

This is a flowchart I have put together for the USA Eastern Interconnection - It represents everything from Manitoba in Canada south to ERCOT (Texas) and all points east to the Atlantic.  It should spatially "resemble" the geography... but can't be perfect.

The "blocks" are the Balancing Authorities (where realtime will hopefully be available).  The links between - all the interconnects (AC or DC) that if available will show as flows.

I will continue working on this... so it would be better if I house it in my Dropbox account and link all that want access.  Just Slack DM me your email address.

**Comment by jarek** [comment]
@HansHyde I appreciate your professional background and your extensive static data, but Electricity Map has been concerning itself with live / real-time emissions since the start, hence my query. There are places to collect information about the world's electrical infrastructure (Energypedia and GENI.org are two that come up in my searches frequently), this issue tracker is probably not the best for it. If you think  there is unlikely to be real-time data available (e.g. for organisational or polical reasons), it's probably not worth opening an issue here.

**Comment by corradio** [comment]
Let's close the issue if we have no leads for interconnection data. Else, let's centralise real-time sources of data here.

Olivier

**Comment by jarek** [comment]
My message was uncalled for and rude. I apologize. Let me rephrase:

Despite the somewhat generic name, Electricity Map is about finding and showcasing real-time CO2eq emissions by geographical region - it finds data for the sister project https://www.co2signal.com/. Consequently it is concerned with real-time or near-real-time generation / fuel mix for a given area.

For a number of mostly non-technical reasons, that data tends to be available mostly by political subdivisions.

Information on system setup, transmission lines, sub-grids, interchanges etc is useful here as far as it helps us calculate real-time emissions (as with NYISO and Ontario IESO being part of Eastern Interconnection but publishing their own data, and us supplementing it with interchange data). Leads for organisations that might have or make generation data available are appreciated.  Beyond that, unfortunately it probably won't help us much.

**Comment by HansHyde** [comment]
@jared, @corradio,

Sorry, I missed your responses.  We appear to be talking past each other right now, not with each other.

I have the framework (which Parsers & all their interconnections) for the entire eastern half of North America, as I outlined above.  The only problem with this... they do not follow state lines. The second issue, the "interconnects" are not like they are in Europe (i.e., DC interconnections), they are more likely multiple different connections (with different real time date sources) which you guys with have to "sum together" to get the Interconnection on ElectricityMap to be correct.

The 26 Balancing Authorities I outline above is our best bet for getting information needed - this is where we will be focusing our efforts to find generation mix & flows to neighboring Balancing Authorities.

So the question I am asking you guys, do you want to mock the entire eastern half of North America from the start, and fill in data as we locate it, or not (and continue trying to chase out rabbit holes with State boundaries)?  Based on @systemcatch work on NEISO, it would appear we could build the remaining Parsers (of the 26 needed) and link to a State geometry for the time being (until customized geometries are available).

You guys decide please, because honestly I feel like I am talking to a wall and wasting my time trying to help you guys out.

**Comment by corradio** [comment]
Hi @HansHyde,
Sorry about not taking the time to outline more clearly how we work (and explaining why we do it that way). Hopefully it will make sense to you and we'll be able to work more efficiently together.

Our process is always:
1. identify a real-time data source (be it production or exchange)
2. figure out if we have an associated geometry (could be a country, a state, or something completely different, like a custom grid geometry)
3. deep dive into the data source to make sure the data is reliable
4. find associated capacities, and update the map

The world is a big place, and starting to map out (i.e. what I believe you refer to as "mock") all zones and exchanges of the world would be an endless task. It would also be quite useless if we don't have associated data.
What we should do instead is focussing on finding real-time data feeds, and mock the related areas as we find those. This will be much more productive, and our conversations and interactions will have a higher signal to noise ratio.

Therefore, I pragmatically will close this issue (as we don't have real-time data sources associated), but the information here-in will be kept nonetheless, as it is very useful once we get real-time data sources for the relevant areas.
We should really focus on #913 in order to incorporate grid geometries (as this will probably happen very soon as we find new data sources). Remember FYI that a good first US area mapping was already done in https://github.com/tmrowco/electricitymap/issues/143#issuecomment-283915610 and that we previously identified https://www.eia.gov/beta/realtime_grid/#/status?end=20171218T01 as a good overview. If issues gets too long, we tend not to read what has been done before, and that's a shame.
@HansHyde if you can help us identify real-time data sources, and leverage your network, that'd be awesome. As I said - mapping out the areas is secondary, and will be quite natural based on where the data comes from.

Let's discuss on Slack if there's anything unclear. Let's keep github issues strictly for tracking progress and adding new information (instead of asking questions indirectly related to the task at hand).

I hope this makes sense. In all cases we're happy to have you here helping us out!

**Comment by HansHyde** [comment]
Copy @corradio - I’ll be back online in about 4 hours.

I’ll add some issues of known areas where we can focus on tracking down info & data streams.

**Comment by corradio** [comment]
Terrific!
